tests/phpunit/tests/test-admin-notices.php

```diff
@@ -9,20 +9,23 @@ public function set_up() {
 		$this->pll_admin = new PLL_Admin( $links_model );
 	}
 
-	public static function wp_redirect() {
-		throw new Exception( 'Call to wp_redirect' );
+	/**
+	 * Allows to continue the execution after wp_redirect + exit.
+	 */
+	protected function filter_wp_redirect() {
+		add_filter(
+			'wp_redirect',
+			function() {
+				throw new Exception( 'Call to wp_redirect' );
+			}
+		);
+
+		$this->expectException( 'Exception' );
+		$this->expectExceptionMessage( 'Call to wp_redirect' );
 	}
 
 	public function test_hide_notice() {
-		// Allows to continue the execution after wp_redirect + exit.
-		add_filter( 'wp_redirect', array( __CLASS__, 'wp_redirect' ) );
-		if ( method_exists( $this, 'setExpectedException' ) ) {
-			// setExpectedException has been deprecated in recent versions of phpunit
-			$this->setExpectedException( 'Exception', 'Call to wp_redirect' );
-		} else {
-			$this->expectException( 'Exception' );
-			$this->expectExceptionMessage( 'Call to wp_redirect' );
-		}
+		$this->filter_wp_redirect();
```

Comment:
```suggestion
		$this->expect_wp_redirect();
```
?

Reply:
Done in 731b9aab50bbf6628cc295fec473c744a768f9b8